#!/usr/bin/env python
PACKAGE = "ipa_rubbish_bin_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("z_min_val", double_t, 0, "Min value in z direction of pass through filter", 0.3, 0.0, 5.0)
gen.add("z_max_val", double_t, 0, "Max value in z direction of pass through filter", 1.4, 0.0, 5.0)

gen.add("num_grasping_points", int_t, 0, "the number of grasping points", 8, 0, 16)


gen.add("cylinder_radius", double_t, 0, "radius search in meters for cylinder detection", 0.25, 0, 1)
gen.add("cylinder_distace_threshold", double_t, 0, "distance threshold for cylinder detection", 0.4, 0, 1.0)

gen.add("cluster_tolerance", double_t, 0, "cluster tolerance", 0.03,0.0,1)
gen.add("min_cluster_size", int_t, 0, "minimum points for euclidean cluster", 10,10,1000)

gen.add("use_remove_outliers", bool_t, 0, "if true, using PCL function to remove outliers", True)


gen.add("floor_plane_inlier_distance", double_t, 0, "Distance threshold value of plane segmentation", 0.02, 0.0, 0.2)
gen.add("min_plane_points", int_t, 0, "minimum number of points that are necessary to find the floor plane", 100, 10, 1000000)
gen.add("floor_search_iterations", int_t, 0, "the number of attempts to segment the floor plane in the image, if set to 0 the parameters of floor_plane_model will be used diectly as plane model without fitting a plane into the 3d data", 3, 0, 100)




gen.add("show_cylinder_marker", bool_t, 0, "if true, visualize cylinder marker in RVIZ", True)
gen.add("show_cylinder_image", bool_t, 0, "if true, visualize bounding box image of cylinder object in RVIZ", True)

#gen.add("sphere_distance_threshold", double_t, 0, "Distance threshold value of sphere segmentation", 0.001, 0.001, 0.1)
#gen.add("circle_distance_threshold", double_t, 0, "Distance threshold value of circle segmentation", 0.001, 0.001, 0.1)

# Second arg is node name it will run in (doc purposes only), third is generated filename prefix
exit(gen.generate(PACKAGE, "ipa_rubbish_bin_detection", "ipa_rubbish_bin_detection"))

